x-pack/plugins/uptime/public/components/fleet_package/combo_box.tsx

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useState, useCallback } from 'react';
+import { EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+
+export interface Props {
+  onChange: (value: string[]) => void;
+  selectedOptions: string[];
+}
+
+export const ComboBox = ({ onChange, selectedOptions }: Props) => {
+  const [formattedSelectedOptions, setSelectedOptions] = useState<
+    Array<EuiComboBoxOptionOption<string>>
+  >(selectedOptions.map((option) => ({ label: option, key: option })));
+  const [isInvalid, setInvalid] = useState(false);
+
+  const onOptionsChange = useCallback(
+    (options: Array<EuiComboBoxOptionOption<string>>) => {
+      setSelectedOptions(options);
+      const formattedTags = options.map((option) => option.label);
+      onChange(formattedTags);
+      setInvalid(false);
+    },
+    [onChange, setSelectedOptions, setInvalid]
+  );
+
+  const onCreateOption = useCallback(
+    (tag: string) => {
+      const formattedTag = tag.trim();
+      const newOption = {
+        label: formattedTag,
+      };
+
+      onChange([...selectedOptions, formattedTag]);
+
+      // Select the option.
+      setSelectedOptions([...formattedSelectedOptions, newOption]);
+    },
+    [onChange, formattedSelectedOptions, selectedOptions, setSelectedOptions]
```

Comment:
i am really surprised that you had to use `useCallBack` so much. with such a long list of dependencies, i am not sure, what performance aspect it provides.

Reply:
and this call back is just setting state object and calling onChange. Which is pretty much what you do in a component.